src/facade/dragonfly_connection.cc

```diff
@@ -288,6 +292,19 @@ void Connection::DispatchOperations::operator()(CheckpointMessage msg) {
   msg.bc.Dec();
 }
 
+void Connection::DispatchOperations::operator()(const InvalidationMessage& msg) {
+  RedisReplyBuilder* rbuilder = (RedisReplyBuilder*)builder;
+  DCHECK(rbuilder->IsResp3());
+  rbuilder->StartCollection(2, facade::RedisReplyBuilder::CollectionType::PUSH);
+  rbuilder->SendBulkString("invalidate");
+  if (msg.invalidate_due_to_flush) {
+    rbuilder->SendNull();
+  } else {
+    std::vector<string_view> keys{msg.key};
+    rbuilder->SendStringArr(keys);
```

Comment:
```c++
string_view[] keys = {msg.key}
SendStringArr(keys)
```
does the same without allocations

Reply:
This doesn't compile

```
/home/yli/git/dragonfly-rebase/src/facade/dragonfly_connection.cc:303:21: error: structured binding declaration cannot have type ‘std::string_view’ {aka ‘std::basic_string_view<char>’}
  303 |     std::string_view[] keys = {msg.key};
      |                     ^~
/home/yli/git/dragonfly-rebase/src/facade/dragonfly_connection.cc:303:21: note: type must be cv-qualified ‘auto’ or reference to cv-qualified ‘auto’
/home/yli/git/dragonfly-rebase/src/facade/dragonfly_connection.cc:303:21: error: empty structured binding declaration
/home/yli/git/dragonfly-rebase/src/facade/dragonfly_connection.cc:303:24: error: expected initializer before ‘keys’
  303 |     std::string_view[] keys = {msg.key};
      |                        ^~~~
/home/yli/git/dragonfly-rebase/src/facade/dragonfly_connection.cc:304:29: error: ‘keys’ was not declared in this scope
  304 |     rbuilder->SendStringArr(keys);
      |                             ^~~~
```

Reply:
try `string_view keys[] = {msg.key};`
It thinks you're using structured binding, using the brackets on the left side is C#/Java syntax

Reply:
this works well, thanks!